### PR TITLE
[Azure Search] Add transform directives for C# generation

### PR DESCRIPTION
**Pull request by bitsforthought**
This applies only to the `Microsoft.Azure.Search.Data` package, the source for which is in azure-sdk-for-net.

We'll be using transform directives more often in the future, so it seemed like a good idea to port our current hacky PowerShell script to this new mechanism. The PowerShell script lives in azure-sdk-for-net, and will be removed in a future PR.

Also did a drive-by fix of an incorrect tag in the searchindex spec. This has no effect on code generation.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by brjohnstmsft** [comment]
FYI @arv100kri

File: specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md [hunk 1]
```diff
@@ -101,3 +101,23 @@ csharp:
...
+Use the below directives sparingly. Every modification we make in here, we potentially have to replicate for every target language.
+
+``` yaml $(csharp)
+directive:
+  # Make all Proxy types internal so we can version them freely.
+  - from: source-file-csharp
+    where: $
+    transform: return $.replace( /public partial interface IDocumentsProxyOperations/g, "internal partial interface IDocumentsProxyOperations" )
+
+  # Change the public property on ISearchIndexClient and SearchIndexClient to refer to our public interface instead of the internal one.
```
Comment (arv100kri) [inline specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md]: Is it worth adding a note, *why* in addition to the *what* ?

**Review by arv100kri — APPROVED** [review]

File: specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md (hunk 1 above)
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md]: There is a broader context behind the _why_, and I'm not sure this is the best place to explain it, but it won't hurt to add a bit more detail.

Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md]: After taking a second look, I realized this transform could be simplified, so I did that too. I'll have a corresponding SDK PR once this has been merged (codegen will fail without it).

**Comment by brjohnstmsft** [comment]
Update: I renamed the operationId prefixes from `DocumentsProxy` to `Documents` because we're planning to start generating REST API documentation from our Swagger, and I didn't want an internal implementation name leaking into the docs. I've updated the transforms accordingly.

**Review by arv100kri — APPROVED** [review]

File: specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md (hunk 1 above)
Comment (arv100kri) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/readme.md]: Makes sense, thanks!

**Comment by brjohnstmsft** [comment]
@sarangan12 Please review & merge

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**